lib/note-content-editor.tsx

```diff
@@ -167,6 +173,86 @@ class NoteContentEditor extends Component<Props> {
     }
   };
 
+  completionProvider: (
+    selectedNoteId: T.EntityId | null
+  ) => languages.CompletionItemProvider = (selectedNoteId) => {
+    return {
+      triggerCharacters: ['['],
+
+      provideCompletionItems(model, position, context, token) {
+        const line = model.getLineContent(position.lineNumber);
+        if (!line.includes('[')) {
+          return null;
+        }
+
+        const precedingOpener = line.lastIndexOf('[', position.column);
+        const precedingCloser = line.lastIndexOf(']', position.column);
+        const precedingBracket =
+          precedingOpener >= 0 && precedingCloser < precedingOpener
+            ? precedingOpener
+            : -1;
+
+        const soFar =
+          precedingBracket >= 0
+            ? line.slice(precedingBracket + 1, position.column)
+            : '';
+
+        const notes = searchNotes(
+          {
+            searchTerms: getTerms(soFar),
+            excludeIDs: selectedNoteId ? [selectedNoteId] : [],
+            openedTag: null,
+            showTrash: false,
+            searchTags: tagsFromSearch(soFar),
+            titleOnly: true,
+          },
+          5
+        ).map(([noteId, note]) => ({
+          noteId,
+          content: note.content,
+          isPinned: note.systemTags.includes('pinned'),
+          ...noteTitleAndPreview(note),
+        }));
+
+        const additionalTextEdits =
+          precedingBracket >= 0
+            ? [
+                {
+                  text: '',
+                  range: {
+                    startLineNumber: position.lineNumber,
+                    startColumn: precedingBracket,
+                    endLineNumber: position.lineNumber,
+                    endColumn: position.column,
+                  },
+                },
+              ]
+            : [];
+
+        return {
+          incomplete: true,
+          suggestions: notes.map((note, index) => ({
+            additionalTextEdits,
+            kind: note.isPinned
+              ? languages.CompletionItemKind.Snippet
+              : languages.CompletionItemKind.File,
```

Comment:
do we all agree that this is right here? I put it in originally as I was exploring the API but I never had strong feelings about it, or if we should try and do anything better with the icons

Reply:
I think we can do something better with it in a follow-up 🤷